metal/datasource_metal_facility.go

```diff
@@ -17,6 +17,13 @@ func dataSourceMetalFacility() *schema.Resource {
 				Description: "The code of the Facility to match",
 				Required:    true,
 			},
+			"desired_features": {
```

Comment:
My first thought was that we should call this field `features`, but then I see that we have the existing `features` attribute. Hmm.

Looking at the description, "which the facility **needs to have**." `required_features` maybe?

What are your thoughts on using `features` with optional + computed?
I've done this before here: https://registry.terraform.io/providers/linode/linode/latest/docs/data-sources/domain_record (and in domain).

I keep running into examples on DO, Linode, and AWS with a `filter { name = "" values = "" }` pattern. We can implement that for arbitrary server-side filtering (when the API supports it). I've opened https://github.com/equinix/terraform-provider-metal/issues/180 to address this.  

When a field is not API filterable, the Terraform argument field name tends to either match the attribute field name or include the name of the filtering operation in the name: `"name_regex" ... "uses local filtering, not available in the API, to allow for more flexible filtering"`.  In this case, the name might be `features_include` or `features_required`.

Reply:
`features_required` sounds good.

> What are your thoughts on using features with optional + computed?

I think it would be messy. It makes the Computed aspect of the field really hard to use. There's no Copmuted + Optional field in the domain record datasource:
https://github.com/linode/terraform-provider-linode/blob/main/linode/data_source_linode_domain_record.go

Reply:
That's interesting. `name` doesn't say optional + computed, but it is treated that way:

https://github.com/linode/terraform-provider-linode/blob/main/linode/data_source_linode_domain_record.go#L88-L117

d.Set is not error-checked there, I suppose this could be erroring.